github/code-scanning.go

```diff
@@ -119,6 +119,16 @@ type AlertListOptions struct {
 	ListOptions
 }
 
+// AnalysisResults specifies the results of a code scanning job.
```

Comment:
Let's please change the name of this to `SarifAnalysis` and add the 3 missing fields found in the documentation pointed to on line 124:

* `checkout_uri`
* `started_at`
* `tool_name`

The, let's please move this entire struct down to right before the new `UploadSarif` method and also use it as one of the parameters for uploading the analysis results.